webapi/src/main/scala/org/knora/webapi/util/ConstructResponseUtilV2.scala

```diff
@@ -426,7 +427,19 @@ object ConstructResponseUtilV2 {
                         textValObj.assertions(OntologyConstants.KnoraBase.ValueHasMapping)
                 }
 
-                acc ++ mappings
+                // get mappings from linked resources
+                val mappingsFromReferredResources: Set[IRI] = valObjs.filter {
```

Comment:
@benjamingeer Is it ok to have a filter and them iterate over the elements that were returned or should I rewrite is using `foldLeft`.

In the lambda function passed to `foldLeft` I could check if the condition is fulfilled and then add to the accumulator or just return the accumulator if not.

Reply:
“Premature optimization is the root of all evil.” --Donald Knuth

`foldLeft` is more efficient than `filter`, but with a small amount of data, the difference might not be measurable. If you’re worried, time it and see.

Reply:
It was more about style. I will leave like this then. :-)